dev/AnimatedVisualPlayer/TestUI/AnimatedVisualPlayerPage.xaml.cs

```diff
@@ -12,45 +13,70 @@
 using Windows.UI;
 using Windows.UI.Composition;
 using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Controls;
 using Windows.UI.Xaml.Hosting;
 using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Tests.MUXControls.ApiTests;
 
 namespace MUXControlsTestApp
 {
-    public class FallbackGrid : Windows.UI.Xaml.Controls.Grid
+    public sealed class FallbackGrid : Grid
     {
+        internal const int RectangleWidth = 100;
+        internal const int RectangleHeight = 100;
+        internal static readonly Color RectangleColor = Colors.Red;
+        // Gets the most recently created FallbackGrid
+        internal static FallbackGrid Latest;
+
         public FallbackGrid()
         {
-            _fallbackGrid = this;
-
+            Latest = this;
+            //Background = new SolidColorBrush(Colors.Red);
+            //this.HorizontalAlignment = HorizontalAlignment.Stretch;
+            //this.VerticalAlignment = VerticalAlignment.Stretch;
             Windows.UI.Xaml.Shapes.Rectangle rect = new Windows.UI.Xaml.Shapes.Rectangle();
-            rect.MinWidth = 100;
-            rect.MinHeight = 100;
-            rect.MaxWidth = 100;
-            rect.MaxHeight = 100;
-            rect.Fill = new SolidColorBrush(Colors.Red);
+            rect.MaxWidth = rect.MinWidth = RectangleWidth;
+            rect.MaxHeight = rect.MinHeight = RectangleHeight;
+            rect.Fill = new SolidColorBrush(RectangleColor);
             Children.Add(rect);
         }
-
-        static public FallbackGrid _fallbackGrid = null;
     }
 
-    /// <summary>
-    /// An empty page that can be used on its own or navigated to within a Frame.
-    /// </summary>
     [TopLevelTestPage(Name = "AnimatedVisualPlayer", Icon = "Animations.png")]
     public sealed partial class AnimatedVisualPlayerPage : TestPage
     {
-        private Visual _visual;
-
         public AnimatedVisualPlayerPage()
         {
             this.InitializeComponent();
+            SetAutomationPeerNames(this);
+        }
+
+        // Set the AutomationPeer name to be the same as the x:Name on each FrameworkElement.
+        // This is so we don't need to set both properties on the elements in XAML.
+        void SetAutomationPeerNames(FrameworkElement root)
```

Comment:
Normally, VisualTreeHelper would drill down into the control template parts of the controls that are on the page, which would not be what is desired. However, because the templates are expanded asynchronously after being loaded and this function is called directly from the Page's .ctor, you get the correct behavior. It is probably worth calling this out in a comment, that it should only be called from the .ctor.

Reply:
An alternative would be to recursively traverse Panel.Children. This would be less general purpose, but it would mean you avoid walking into the parts of an expanded ControlTemplate.

---
In reply to: [343822102](https://github.com/microsoft/microsoft-ui-xaml/pull/1558#discussion_r343822102) [](ancestors = 343822102)

Reply:
Done.